diff --git a/CMakeLists.txt b/CMakeLists.txt
index b7b98838..03798898 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,7 +143,7 @@ function(remove_component_from_property property_name component_to_remove)
 endfunction()
 
 # Disable warnings that show up in external code (gtest;pybind11)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-covered-switch-default -fvisibility=hidden")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-covered-switch-default -fvisibility=hidden -Wno-deprecated-declarations")
 
 include_directories(${TRITON_ROOT_DIR})
 include_directories(${MLIR_INCLUDE_DIRS})
diff --git a/ascend/triton-adapter/safe_compile.cmake b/ascend/triton-adapter/safe_compile.cmake
index 2b29614c..04a256c4 100644
--- a/ascend/triton-adapter/safe_compile.cmake
+++ b/ascend/triton-adapter/safe_compile.cmake
@@ -5,6 +5,6 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector-strong")
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,now -pie -s")
 set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,now -s")
-set(CMAKE_SKIP_RPATH TRUE)
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE)
-unset(CMAKE_INSTALL_RPATH)
\ No newline at end of file
+# set(CMAKE_SKIP_RPATH TRUE)
+# set(CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE)
+# unset(CMAKE_INSTALL_RPATH)
\ No newline at end of file
diff --git a/setup.py b/setup.py
index 56819b4e..dcdff93f 100644
--- a/setup.py
+++ b/setup.py
@@ -759,7 +759,7 @@ setup(
     packages=get_packages(_backends),
     package_data=get_package_data(_backends),
     include_package_data=True,
-    ext_modules=[CMakeExtension("triton", "triton/_C/")],
+    # ext_modules=[CMakeExtension("triton", "triton/_C/")],
     cmdclass={
         "build_ext": BuildExt,
         "install": BuildInstall,
diff --git a/.gitmodules b/.gitmodules
index b3d4aefb..a31421f5 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,6 +1,6 @@
 [submodule "third_party/triton"]
 	path = third_party/triton
-	url = https://github.com/triton-lang/triton.git
+	url = git@github.com:triton-lang/triton.git
 	depth = 1
 [submodule "third_party/ascendnpu-ir"]
 	path = third_party/ascendnpu-ir
